src/custom/components/Popups/FollowPendingTxPopup/index.tsx

```diff
@@ -9,38 +9,95 @@ import {
   handleFollowPendingTxPopupAtom,
   handleHidePopupPermanentlyAtom,
   showFollowTxPopupAtom,
+  followPendingTxPopupAtom,
+  selectAtom,
+  handleCloseOrderPopupAtom,
 } from 'state/application/atoms'
 import FollowPendingTxPopupUI from './FollowPendingTxPopupUI'
 import { isPending } from 'components/Web3Status'
+import { OrderID } from '@src/custom/api/gnosisProtocol'
 
-function isAnOrder(element: TransactionAndOrder): element is Order & { addedTime: number } {
+interface AddedOrder extends Order {
+  addedTime: number
+}
+
+function isAnOrder(element: TransactionAndOrder): element is AddedOrder {
   return 'inputToken' in element && 'outputToken' in element
 }
 
-const FollowPendingTxPopup: React.FC = ({ children }): JSX.Element => {
+export const useLastPendingOrder = (): { lastPendingOrder: AddedOrder | null; onClose: () => void } => {
   const setShowFollowPendingTxPopup = useUpdateAtom(handleFollowPendingTxPopupAtom)
-  const setHidePendingTxPopupPermanently = useUpdateAtom(handleHidePopupPermanentlyAtom)
-  const showFollowPendingTxPopup = useAtomValue(showFollowTxPopupAtom)
-  const isExpertMode = useIsExpertMode()
+  const setLastOrderClosed = useUpdateAtom(handleCloseOrderPopupAtom)
   const allRecentActivity = useRecentActivity()
-  const isTherePendingOrder = useMemo(() => {
-    if (!isExpertMode) return
+  const lastPendingOrder = useMemo(() => {
     const pendings = allRecentActivity.filter(isPending)
+    const lastOrder = pendings[pendings.length - 1]
+
+    if (!lastOrder || !isAnOrder(lastOrder)) return null
+
+    return lastOrder
+  }, [allRecentActivity])
+
+  const onClose = useCallback(() => {
+    lastPendingOrder && setLastOrderClosed(lastPendingOrder.id)
+    setShowFollowPendingTxPopup(false)
+  }, [lastPendingOrder, setLastOrderClosed, setShowFollowPendingTxPopup])
+
+  return { lastPendingOrder, onClose }
+}
+
+// Set pop up closed if it has not been closed and not fulfill a condition such as not pending tx
+export function useCloseFollowTxPopupIfNotPendingOrder() {
+  const showingPopup = useAtomValue(showFollowTxPopupAtom)
+  const { lastPendingOrder, onClose } = useLastPendingOrder()
+  const onCloseRef = useRef<() => void>()
+
+  useEffect(() => {
+    if (lastPendingOrder && showingPopup) {
+      onCloseRef.current = onClose
+    } else if (!lastPendingOrder && showingPopup && onCloseRef.current) {
+      onCloseRef.current()
+    }
+  }, [lastPendingOrder, onClose, showingPopup])
+}
 
-    return isAnOrder(pendings[pendings.length - 1])
-  }, [allRecentActivity, isExpertMode])
+const useShowingPopupFirstTime = (orderId: OrderID | undefined) => {
+  const showingPopup = useAtomValue(showFollowTxPopupAtom)
+  const _orderPopupHasBeenClosed = useMemo(
+    () =>
+      selectAtom(followPendingTxPopupAtom, ({ lastOrderPopupClosed }) => {
+        if (!orderId) return
+
+        return lastOrderPopupClosed === orderId
```

Comment:
If `!orderId` is undefined, wouldn't that return false when comparing with `lastOrderPopupClosed`?

I see that `lastOrderPopupClosed` can be `null` and that will be a problem when both are falsy.

And for that I ask: why `lastOrderPopupClosed` defaults to `null` rather than `undefined`?

Not a big deal, just trying understand if there's any strong reason for it

Reply:
When I started I thought it was necessary to compare the initial state, undefined by default is best.